hugegraph-api/src/main/java/com/baidu/hugegraph/serializer/JsonSerializer.java

```
@@ -85,7 +85,13 @@ public String writeList(String label, Collection<?> list) {
     private String writeIterator(String label, Iterator<?> iter,
                                  boolean paging) {
         // Early throw if needed
-        iter.hasNext();
+        try{
+            iter.hasNext();
+        }catch (NullPointerException e){
+            e.printStackTrace();
+        }catch (RuntimeException e){
+            e.printStackTrace();
+        }
```

IMO, we can't add the catch block and ignore the exception silently.

---

增加预分区feature 后 运行hugecomputer 发现 扫描数据发生了NPE，后端hugeserver正常运行，当时通过后端日志以及客户端抛出的NPE 很难定位问题 最后通过DEBUG 跟到代码最外层 通过try{}catch{} 后端输出了异常堆栈 如下：
`java.lang.RuntimeException: java.lang.NullPointerException
        at org.apache.hadoop.hbase.client.RpcRetryingCallerImpl.callWithoutRetries(RpcRetryingCallerImpl.java:201)
        at org.apache.hadoop.hbase.client.ClientScanner.call(ClientScanner.java:266)
        at org.apache.hadoop.hbase.client.ClientScanner.loadCache(ClientScanner.java:434)
        at org.apache.hadoop.hbase.client.ClientScanner.nextWithSyncCache(ClientScanner.java:309)
        at org.apache.hadoop.hbase.client.ClientScanner.next(ClientScanner.java:594)
        at org.apache.hadoop.hbase.client.ResultScanner$1.hasNext(ResultScanner.java:53)
        at com.baidu.hugegraph.backend.store.hbase.HbaseSessions$RowIterator.hasNext(HbaseSessions.java:920)
        at com.baidu.hugegraph.backend.serializer.BinaryEntryIterator.fetch(BinaryEntryIterator.java:71)
        at com.baidu.hugegraph.backend.store.BackendEntryIterator.hasNext(BackendEntryIterator.java:59)
        at com.baidu.hugegraph.iterator.ListIterator.<init>(ListIterator.java:37)
        at com.baidu.hugegraph.backend.query.QueryResults.toList(QueryResults.java:173)
        at com.baidu.hugegraph.backend.query.QueryResults.toList(QueryResults.java:91)
        at com.baidu.hugegraph.backend.page.QueryList$OptimizedQuery.iterator(QueryList.java:178)
        at com.baidu.hugegraph.backend.page.QueryList.fetchNext(QueryList.java:121)
        at com.baidu.hugegraph.backend.page.PageEntryIterator.fetch(PageEntryIterator.java:80)
        at com.baidu.hugegraph.backend.page.PageEntryIterator.hasNext(PageEntryIterator.java:66)
        at com.baidu.hugegraph.iterator.FlatMapperIterator.fetch(FlatMapperIterator.java:58)
        at com.baidu.hugegraph.iterator.WrappedIterator.hasNext(WrappedIterator.java:40)
        at com.baidu.hugegraph.iterator.FilterIterator.fetch(FilterIterator.java:42)
        at com.baidu.hugegraph.iterator.WrappedIterator.hasNext(WrappedIterator.java:40)
        at com.baidu.hugegraph.backend.tx.GraphTransaction$LimitIterator.fetch(GraphTransaction.java:2009)
        at com.baidu.hugegraph.iterator.WrappedIterator.hasNext(WrappedIterator.java:40)
        at com.baidu.hugegraph.serializer.JsonSerializer.writeIterator(JsonSerializer.java:93)
        at com.baidu.hugegraph.serializer.JsonSerializer.writeEdges(JsonSerializer.java:224)
        at com.baidu.hugegraph.api.traversers.EdgesAPI.scan(EdgesAPI.java:123)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144)
        at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161)
        at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205)`
增加此段异常捕捉目的是能够快速发现异常原因，目前问题已经定位 此段代码也可以不加 @javeme @imbajin 辛苦帮忙再review下

---

there is an option `exception.allow_trace` to debug issues, we can enable it when needed: https://hugegraph.github.io/hugegraph-doc/config/config-option.html